### PR TITLE
ETH-781 - referral in history

### PR DESCRIPTION
**Pull request by eduardmaximovich**
## Jira Ticket

https://linear.app/etherean/issue/ETH-781/[android]-🍕referral-reward-v-istorii-tranzakcij

## Description of Work

[ what work was done in this ticket ]

## Screenshots (Optional)

| Screenshot #1 | Screenshot #2 | Screenshot #3 |
|:-------------:|:-------------:|:-------------:|
| [Paste Pic 1] | [Paste Pic 2] | [Paste Pic 3] |

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/history/ui/detailsbottomsheet/HistoryTransactionDetailsPresenter.kt [hunk 1]
```diff
@@ -28,13 +33,23 @@ class HistoryTransactionDetailsPresenter(
     HistoryTransactionDetailsContract.Presenter {
 
     private val timeFormat = DateTimeFormatter.ofPattern(DateTimeUtils.PATTERN_FULL_DAY, Locale.US)
+    private var tx: HistoryTransaction? = null
+    private var loadUsernameJob: Job? = null
+
+    override fun detach() {
+        super.detach()
+        loadUsernameJob?.cancel()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/history/ui/detailsbottomsheet/HistoryTransactionDetailsPresenter.kt:41]: won't be it cancelled automatically when the root scope is canceled?

File: app/src/main/java/org/p2p/wallet/history/ui/detailsbottomsheet/HistoryTransactionDetailsPresenter.kt [hunk 2]
```diff
@@ -325,4 +394,22 @@ class HistoryTransactionDetailsPresenter(
             )
         }
     }
+
+    override fun onFirstButtonClick() {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/history/ui/detailsbottomsheet/HistoryTransactionDetailsPresenter.kt:398]: Maybe Primary and Secondary?

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/history/ui/detailsbottomsheet/HistoryTransactionDetailsPresenter.kt (hunk 1 above)
Comment (eduardmaximovich) [reply app/src/main/java/org/p2p/wallet/history/ui/detailsbottomsheet/HistoryTransactionDetailsPresenter.kt:41]: ah I always forgout about, you're right
